app/assets/javascripts/discourse/tests/acceptance/user-menu-test.js

```diff
@@ -187,6 +189,33 @@ acceptance("User menu", function (needs) {
     );
   });
 
+  test("messages tab applies model transformations registered by plugins", async function (assert) {
+    withPluginApi("0.1", (api) => {
+      api.registerModelTransformer("topic", (topics) => {
+        topics.forEach((topic) => {
+          topic.fancy_title = `pluginTransformer#1 ${topic.fancy_title}`;
+        });
+      });
+      api.registerModelTransformer("topic", async (topics) => {
+        // sleep 10 ms
+        await new Promise((resolve) => later(resolve, 10));
```

Comment:
I'm curious why we need this line here.

Reply:
It tests that `async` callbacks are correctly `await`ed. If we were to refactor how this API is implemented and during the refactor an `await` was accidentally dropped from somewhere or a `for` loop was refactored to `forEach`, then this test would fail and alert us to the problem.

Reply:
If that is the case, does it have to be 10ms? Wondering if we can just return an empty promise.

Reply:
Empty promise breaks the test, but it looks like we can lower it to 1ms which I've done.